internal/resolution/manifest/maven.go

```diff
@@ -329,6 +330,196 @@ func mavenOrigin(list ...string) string {
 	return result
 }
 
+func (MavenManifestIO) Write(df lockfile.DepFile, w io.Writer, patch ManifestPatch) error {
+	specific, ok := patch.EcosystemSpecific.(MavenManifestSpecific)
+	if !ok {
+		return errors.New("invalid MavenManifestSpecific data")
+	}
+	depPatches, propertyPatches, err := buildPatches(patch.Deps, specific)
+	if err != nil {
+		return err
+	}
+
+	buf := new(bytes.Buffer)
+	if _, err := buf.ReadFrom(df); err != nil {
+		return fmt.Errorf("failed to read from DepFile: %w", err)
+	}
+
+	return write(buf, w, depPatches, propertyPatches)
+}
+
+type MavenPatch struct {
+	maven.DependencyKey
+	NewRequire string
+}
+
+// MavenDependencyPatches represent the dependencies to be updated, which
+// is a map of dependency patches of each origin.
+type MavenDependencyPatches map[string]map[MavenPatch]bool // origin -> patch -> whether from base project
+
+// addPatch adds a patch to the patches map indexed by origin.
+// fromBase indicates whether this patch comes from the base project.
+func (m MavenDependencyPatches) addPatch(changedDep DependencyPatch, fromBase bool) error {
+	d, o, err := resolve.MavenDepTypeToDependency(changedDep.Type)
+	if err != nil {
+		return fmt.Errorf("MavenDepTypeToDependency: %w", err)
+	}
+
+	substrings := strings.Split(changedDep.Pkg.Name, ":")
+	if len(substrings) != 2 {
+		return fmt.Errorf("invalid Maven name: %s", changedDep.Pkg.Name)
+	}
+	d.GroupID = maven.String(substrings[0])
+	d.ArtifactID = maven.String(substrings[1])
+
+	if _, ok := m[o]; !ok {
+		m[o] = make(map[MavenPatch]bool)
+	}
+	m[o][MavenPatch{
+		DependencyKey: d.Key(),
+		NewRequire:    changedDep.NewRequire,
+	}] = fromBase
+
+	return nil
+}
+
+// MavenPropertyPatches represent the properties to be updated, which
+// is a map of properties of each origin.
+type MavenPropertyPatches map[string]map[string]string // origin -> tag -> value
+
+// buildPatches returns dependency patches ready for updates.
+func buildPatches(patches []DependencyPatch, specific MavenManifestSpecific) (MavenDependencyPatches, MavenPropertyPatches, error) {
+	depPatches := MavenDependencyPatches{}
+	propertyPatches := MavenPropertyPatches{}
+	for _, patch := range patches {
+		origDep := originalDependency(patch, specific.OriginalRequirements)
+		if origDep.Name() == "" {
+			// An empty name indicates the dependency is not found, so the original dependency is not in the base project.
+			// TODO: enable for guided remeidation
+			// if err := depPatches.addPatch(patch, false); err != nil {
+			//	return MavenDependencyPatches{}, MavenPropertyPatches{}, err
+			// }
+			continue
+		}
```

Comment:
Is there a particular reason to have this disabled?

Reply:
currently patches from both guided remediation and general updates do not set `OrigRequire` which makes it hard to differentiate them (we don't want to add patches unfound in base project if they are for general updates) and there is also a TODO for that purpose.